### DISCUSSION TITLE
Merge of Multiblocks that have partial values

### DISCUSSION THREAD

**Discussion by npillardou** [pyvista/pyvista#5704] [Q&A] [answered]
Dear Pyvista community, 

I am currently working on a project where I am dealing with Multiblocks pvd file.

In each block, some values are partial, because they are coming from different solvers. 

I am trying to merge them in order to do then a sample over line on the whole domain. 

Do you think there exists a possibility to merge them, even the partial values and extend them directly to the whole domain ? 

Because the actual merging blocks functions only keep values that are in common in each block. 

Thank you in advance for your answer.

Regards,

Nicolas

**Comment by MatthewFlamm** [comment]
I think the answer will depend on exactly the situation.  Do you have the same mesh, but different blocks that have different arrays?  Do you have different mesh pieces that are separate each with different arrays?  Are they overlapping meshes with different structure?

**Reply by npillardou** [reply]
Thank you Matthew for your answer.

Technically, I have two different meshes and each of them have some values that are common on both, but some properties have different names due to two different solvers used to solve a flow problem. 

May be I can deal with two different sample over line and join them thanks to coordinates. 

But I would like to know if there exists a advanced merge function that add also properties which are not in common in each mesh. 

Thank you

**Reply by MatthewFlamm** [reply, marked as answer]
Some suggestions: 1) do `sample_over_line` for each mesh and then combine in a suitable way like you suggest. 2) add data arrays to all pieces with a suitable value for non-available data, then merge/combine, then `sample_over_line`.

Note that `sample_over_line` is a simple wrapper around `sample` filter.  See https://github.com/pyvista/pyvista/blob/release/0.43/pyvista/core/filters/data_set.py#L4313-L4370.  The `vtkGhostType` retrun will be useful here for the 1st possibility, in that it will mark portions of the line that aren't within each mesh.

**Reply by npillardou** [reply]
Thank you for your help Matthew, I finally managed to do it.
